### PR TITLE
Theming Structure + Helper Functions + Settings Page

### PR DESCRIPTION
**Pull request by acnebs**
- Fix React/Redux devtools not being loaded in development.
- Modal should only exist in one part of the hierarchy.
- Add a generic `<Select />` component which matches the app theme and has snazzy animations.
- Introduce theming support with a variety of helper functions to make it easy / maintainable.
- Add a settings page
  - Mostly for settings that don't exist yet
  - ~Is where auto-updates / checking for updates is managed~
- ~Working Github Action for publishing distributable sautomatically when pre-releases are created.~

### CODE REVIEW

**Comment by acnebs** [comment]
Not ready to merge yet as still putting finishing touches on the Settings page.

**Comment by piggydoughnut** [comment]
It would be great if you could write a list of changes in the description too 😉

File: src/components/Home/Home.tsx [hunk 1]
```diff
@@ -48,19 +52,21 @@ const renderView = viewType => {
...
 
 const Home = () => {
   const currentView = useSelector(selectView);
+  const modalProps = modals[useSelector(selectModal)];
 
   return (
     <HomeRoot>
       <SideMenu />
       <ViewWrapper>{renderView(currentView)}</ViewWrapper>
+      {modalProps && <Modal title={modalProps.title}>{modalProps.children}</Modal>}
```
Comment (piggydoughnut) [inline src/components/Home/Home.tsx:69]: Oh! that is great :)

File: src/util/theming.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
+import { css } from '@emotion/react';
+import { isObject, merge, omit } from 'lodash';
+
+// settings
+const ignoredKeys = ['name'];
+
+// themes
+const baseTheme = {
+  color: {
+    cornflower: 'hsla(231, 100%, 69%, 1.0)',
```
Comment (piggydoughnut) [inline src/util/theming.ts:10]: love the usage of hsla ❤️

File: src/components/Settings/Settings.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  ${Responsive.below.L} {
+    width: 100%;
+  }
+  margin-bottom: 28px;
+  padding: 0 14px;
+`;
+
+const SubsectionHeader = styled.div`
+  margin-bottom: 16px;
+  font-size: 20px;
```
Comment (piggydoughnut) [inline src/components/Settings/Settings.tsx:28]: Please adjust headers according to design (h2, h3 sizes). Not sure why the headers are divs?

**Comment by piggydoughnut** [comment]
Please adjust according to the design

Screen Shot 2021-06-21 at 5 05 15 PM

Also I see some new console issues, in addition to existing ones  :)

Screen Shot 2021-06-21 at 4 50 43 PM

File: package.json [hunk 4]
```diff
@@ -9,7 +9,7 @@
     "tsc": "cross-env NODE_ENV=development ./node_modules/.bin/tsc",
     "checks": "concurrently 'yarn lint' 'yarn test' 'yarn tsc'",
     "postinstall": "node -r @babel/register ./config/scripts/CheckNativeDep.js && electron-builder install-app-deps && yarn cross-env NODE_ENV=development webpack --config ./config/webpack/webpack.config.renderer.dev.dll.babel.js && yarn-deduplicate yarn.lock",
-    "start": "NODE_ENV=production node -r @babel/register ./config/scripts/CheckPortInUse.js && cross-env yarn start:renderer",
+    "start": "NODE_ENV=production env-cmd -f env/electron-publish.env node -r @babel/register ./config/scripts/CheckPortInUse.js && cross-env yarn start:renderer",
```
Comment (piggydoughnut) [inline package.json:12]: so is the file `.env` or `env/electron-publish.env`. Is `env/electron-publish.env` create by webpack or smth? I would add some reference to the `dev` command below too.

Comment (piggydoughnut) [reply package.json:12]:
```
➜  app git:(visual_adjustments) ✗ yarn start
yarn run v1.22.10
$ NODE_ENV=production env-cmd -f env/electron-publish.env node -r @babel/register ./config/scripts/CheckPortInUse.js && cross-env yarn start:renderer
Error: Failed to find .env file at path: env/electron-publish.env
    at getEnvFile (/Users/dariah/Documents/Dev/tokel/app/node_modules/env-cmd/dist/get-env-vars.js:40:19)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by acnebs** [comment]
Removed the auto-update/release functionality for the time being as getting it to work alongside the current `libnspv` system was proving far too involved.

**Pull request merged**